tests/unit/convert_gs1_xml_to_json.t

```diff
@@ -0,0 +1,74 @@
+#!/usr/bin/perl -w
+
+use Modern::Perl '2017';
+use utf8;
+
+use JSON;
+
+use Test::More;
+use Test::Number::Delta relative => 1.001;
+use Log::Any::Adapter 'TAP';
+
+use ProductOpener::GS1 qw/:all/;
+use ProductOpener::Test qw/:all/;
+
+use JSON "decode_json";
+
+my ($test_id, $test_dir, $expected_result_dir, $update_expected_results) = (init_expected_results(__FILE__));
+
+my $input_dir = "$test_dir/inputs/$test_id";
+
+opendir(my $dh, $input_dir) or die("Could not open the $input_dir directory: $!\n");
+
+foreach my $file (sort(readdir($dh))) {
+
+	next if $file !~ /\.xml$/;
+	my $file = $`;    # remove xml extension
+	my $json_file = "$input_dir/$file.json";
+
+	convert_gs1_xml_file_to_json("$input_dir/$file.xml", $json_file);
+
+	if (open(my $json_fh, "<:encoding(UTF-8)", $json_file)) {
+		local $/;    #Enable 'slurp' mode
+		my $json = JSON->new->allow_nonref->canonical;
+		my $json_ref = $json->decode(<$json_fh>);
```

Comment:
Personally instead of using slurp mode, I prefer to use join:
```suggestion
		my $json = JSON->new->allow_nonref->canonical;
		my $json_ref = $json->decode(join("", <$json_fh>));
```